.github/workflows/_run_e2e_tests.yml

```diff
@@ -0,0 +1,179 @@
+          OPEN_AI_KEY: "${{ secrets.E2E_TESTS_OPEN_AI_KEY }}"
+          AZURE_OPEN_AI_KEY: "${{ secrets.E2E_TESTS_AZURE_OPEN_AI_KEY }}"
+          AZURE_OPEN_AI_ENDPOINT: "${{ secrets.E2E_TESTS_AZURE_OPEN_AI_ENDPOINT }}"
+          AZURE_BLOB_STORAGE_CONNECTION_STRING: "${{ secrets.E2E_TESTS_AZURE_BLOB_STORAGE_CONNECTION_STRING }}"
+          GCLOUD_ACCOUNT_KEY_JSON: "${{ secrets.E2E_TESTS_GCLOUD_ACCOUNT_KEY_JSON }}"
+          GOOGLE_API_KEY: "${{ secrets.E2E_TESTS_GOOGLE_API_KEY }}"
+          AWS_ACCESS_KEY_ID: "${{ secrets.E2E_TESTS_AWS_ACCESS_KEY_ID }}"
+          AWS_SECRET_ACCESS_KEY: "${{ secrets.E2E_TESTS_AWS_SECRET_ACCESS_KEY }}"
+          BEDROCK_AWS_REGION: "${{ secrets.E2E_TESTS_BEDROCK_AWS_REGION }}"
+          HUGGINGFACE_HUB_KEY: "${{ secrets.E2E_TESTS_HUGGINGFACE_HUB_KEY }}"
+          NVIDIA_API_KEY: "${{ secrets.E2E_TESTS_NVIDIA_API_KEY }}"
+          LANGCHAIN_API_KEY: "${{ secrets.E2E_TESTS_LANGCHAIN_API_KEY }}"
+        run: |
+          source scripts/ci-common-env.sh
+          if [ "${{ inputs.suite-name == 'ragstack' }}" == "true" ]; then
+            tox -c ragstack-e2e-tests
+          elif [ "${{ inputs.suite-name == 'langchain' }}" == "true" ]; then
+            tox -c ragstack-e2e-tests -e langchain
+          elif [ "${{ inputs.suite-name == 'llamaindex' }}" == "true" ]; then
+            tox -c ragstack-e2e-tests -e llamaindex
+          elif [ "${{ inputs.suite-name == 'ragstack-latest-release' }}" == "true" ]; then
+            tox -c ragstack-e2e-tests -e ragstack-ai-latest
+          else
+            echo "Unknown suite name: ${{ inputs.suite-name }}"
+            exit 1
+          fi
+
+      - name: Dump report on Github Summary
+        if: ${{ always() }}
+        uses: ./.github/actions/add-report-to-github-summary
+
+      - name: Compute info
+        if: ${{ always() }}
+        id: info
+        run: |
+          if [ "${{ inputs.suite-name == 'ragstack' }}" == "true" ]; then
+            commit_ref=$(git rev-parse --short HEAD)
+            echo "commit-url=https://github.com/datastax/ragstack-ai/commits/${commit_ref}" >> $GITHUB_OUTPUT
+          
+          elif [ "${{ inputs.suite-name == 'langchain' }}" == "true" ]; then
+            commit_ref=$(grep resolved_reference ragstack-e2e-tests/poetry.lock | awk -F'"' '{print $2}' | head -1)
+            echo "commit-url=https://github.com/langchain-ai/langchain/commits/${commit_ref}" >> $GITHUB_OUTPUT
+
+          elif [ "${{ inputs.suite-name == 'llamaindex' }}" == "true" ]; then
+            commit_ref=$(grep resolved_reference ragstack-e2e-tests/poetry.lock | awk -F'"' '{print $2}' | head -1)
+            echo "commit-url=https://github.com/run-llama/llama_index/commits/${commit_ref}" >> $GITHUB_OUTPUT
+
+          elif [ "${{ inputs.suite-name == 'ragstack-latest-release' }}" == "true" ]; then
+            (tox run -c ragstack-e2e-tests -e get-latest-ragstack-ai-version -q) | tail -3 | head -n 1 > /tmp/ragstack-ai-version
```

Comment:
hm, why does this need the `tail | head`?

Reply:
we want the last 3rd line. it's weird but c'est la vie